lib/python/rose/config_processors/fileinstall.py

```diff
@@ -558,11 +562,14 @@ def __init__(self, name, checksum=None, access_mode=None):
         self.checksum = checksum
         self.access_mode = access_mode
 
-    def __cmp__(self, other):
+    def __lt__(self, other):
         return (
-            cmp(self.name, other.name) or
-            cmp(self.checksum, other.checksum) or
-            cmp(self.access_mode, other.access_mode))
+            (self.name > other.name) -
+            (self.name < other.name) or
+            (self.checksum > other.checksum) -
+            (self.checksum < other.checksum) or
+            (self.access_mode > other.access_mode) -
+            (self.access_mode < other.access_mode))
```

Comment:
Can we just do:

```python
def __lt__(self, other):
    return (
        self.name < other.name
        or self.checksum < other.checksum
        or self.access_mode < other.access_mode
    )
```

Reply:
I'll give this a go.

Reply:
I've used the second. Perhaps we need a unittest for this?
I'd take a straw poll and add it to the list of things to do.

Reply:
A unittest wouldn't hurt.

Reply:
K. - Will do.

Reply:
Spun this off into ticket #2302